ddnoise.js

```diff
@@ -53,17 +53,22 @@ define(['./utils', 'underscore', 'promise'], function (utils, _) {
     };
 
     DdNoise.prototype.oneShot = function (sound) {
-        var source = this.context.createBufferSource();
+        var duration = sound.duration;
+        var context = this.context;
+        if (context.state !== "running") return duration;
+        var source = context.createBufferSource();
         source.buffer = sound;
         source.connect(this.gain);
         source.start();
-        return sound.duration;
+        return duration;
     };
 
     DdNoise.prototype.play = function (sound, loop) {
         var self = this;
+        var context = self.context;
+        if (context.state !== "running") return null;
```

Comment:
It might clean the code up to return a resolved promise here (`return Promise.resolve();`). That would get rid of the `if` check for null on line 91/92.

Reply:
Yes, that was a bit ham-fisted. Updated patch uses Promise.reject() which I think it conceptually what is going on here. (Promise.resolve() seemed to work but I think it may have left the ddnoise object in an inconsistent state.)